src/python/grpcio/grpc/experimental/aio/_interceptor.py

```diff
@@ -0,0 +1,334 @@
+
+    def cancel(self) -> bool:
+        return False
+
+    def cancelled(self) -> bool:
+        return False
+
+    def done(self) -> bool:
+        return True
+
+    def add_done_callback(self, unused_callback) -> None:
+        raise NotImplementedError()
+
+    def time_remaining(self) -> Optional[float]:
+        raise NotImplementedError()
+
+    async def initial_metadata(self) -> Optional[MetadataType]:
+        return None
+
+    async def trailing_metadata(self) -> Optional[MetadataType]:
+        return self._error.initial_metadata()
+
+    async def code(self) -> grpc.StatusCode:
+        return self._error.code()
+
+    async def details(self) -> str:
+        return self._error.details()
+
+    async def debug_error_string(self) -> Optional[str]:
+        return self._error.debug_error_string()
+
+    def __await__(self):
+        raise self._error
+
+
+class UnaryUnaryCallResponse(_base_call.UnaryUnaryCall):
+    """Final UnaryUnaryCall class finished with a response."""
+    _response: ResponseType
+
+    def __init__(self, response: ResponseType) -> None:
+        self._response = response
+
+    def cancel(self) -> bool:
+        return False
+
+    def cancelled(self) -> bool:
+        return False
+
+    def done(self) -> bool:
+        return True
+
+    def add_done_callback(self, unused_callback) -> None:
+        raise NotImplementedError()
+
+    def time_remaining(self) -> Optional[float]:
+        raise NotImplementedError()
+
+    async def initial_metadata(self) -> Optional[MetadataType]:
+        return None
+
+    async def trailing_metadata(self) -> Optional[MetadataType]:
+        return None
+
+    async def code(self) -> grpc.StatusCode:
+        return grpc.StatusCode.OK
+
+    async def details(self) -> str:
+        return ''
+
+    async def debug_error_string(self) -> Optional[str]:
+        return None
+
+    def __await__(self):
+        if False:  # pylint: disable=W0125
+            # This code path is never used, but a yield statement is needed
+            # for telling the interpreterthat __await__ is a generator.
+            yield None
+        return self._response
+
+
+class UnaryUnaryCancelledError(_base_call.UnaryUnaryCall):
+    """Final UnaryUnaryCall class finished with an asyncio.CancelledError."""
+
+    def cancel(self) -> bool:
+        return False
+
+    def cancelled(self) -> bool:
+        return True
+
+    def done(self) -> bool:
+        return True
+
+    def add_done_callback(self, unused_callback) -> None:
+        raise NotImplementedError()
+
+    def time_remaining(self) -> Optional[float]:
+        raise NotImplementedError()
+
+    async def initial_metadata(self) -> Optional[MetadataType]:
+        return None
+
+    async def trailing_metadata(self) -> Optional[MetadataType]:
+        return None
+
+    async def code(self) -> grpc.StatusCode:
+        return grpc.StatusCode.CANCELLED
+
+    async def details(self) -> str:
+        return ''
```

Comment:
nit: If we want to keep this class, we need the details to have some content like "cancelled locally".